src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl

```diff
@@ -35,100 +32,258 @@ export saturation_excess
 export liquid_fraction, saturation_adjustment, phase_partitioning_eq
 
 # Auxiliary functions, e.g., for diagnostic purposes
-export liquid_ice_pottemp, dry_pottemp, exner
+export liquid_ice_pottemp, liquid_pottemp, dry_pottemp, density_pottemp, exner
+export mix_ratio_con, mix_ratio_vap
 
+# Thermodynamic states
+export ThermodynamicState, InternalEnergy_Shum_eq, InternalEnergy_Shum_neq
+export LiqPottemp_Shum_eq
```

Comment:
I'm not sure on the mixed case. How about
```
InternalEnergySHumEquil
InternalEnergySHumNonEquil
LiquidPotTempSHumEquil
```

Reply:
I'd replace `SHumEquil` by `PhaseEquil` or something similar, and same for non-equilibrium (because thermally there still is equilibrium...)

Reply:
And I'm also not sure we need to support liquid (or, better, liquid-ice) potential temperature as a state variable. It's not generally a good choice (for starters, it is not additive). Seems like overloading the code with unnecessary options to me.

Reply:
> I'd replace `SHumEquil` by `PhaseEquil` or something similar, and same for non-equilibrium (because thermally there still is equilibrium...)

This looks like the right abstraction. I've changed `InternalEnergySHumEquil` to `PhaseEquil`, and similar for non-equilibrium.

Reply:
> And I'm also not sure we need to support liquid (or, better, liquid-ice) potential temperature as a state variable. It's not generally a good choice (for starters, it is not additive). Seems like overloading the code with unnecessary options to me.

I think you'll like our new solution. We're accepting these variables as input, but they are immediately converted into a `PhaseEquil` object, using the functions we discussed. So, there are currently only two structs: `PhaseEquil` and `PhaseNonEquil`.